clang/lib/CodeGen/CoverageMappingGen.cpp

```diff
@@ -1808,12 +1808,24 @@ struct CounterCoverageMappingBuilder
     }
   }
 
+private:
+  static bool evaluateConstantCondition(const Expr *Condition) {
+    if (const auto *Expr = dyn_cast<ConstantExpr>(Condition))
+      return Expr->getResultAsAPSInt().getExtValue();
+
+    if (const auto *Expr = dyn_cast<ExprWithCleanups>(Condition))
+      return evaluateConstantCondition(Expr->getSubExpr()); // recursion
+
+    assert(false && "Unexpected node in 'if constexpr' condition");
+    return false;
+  }
+
+public:
```

Comment:
@AaronBallman This is basically an `IgnoreImpCast` except we want to stop at the first ConstantExpr (which is a FullExpr) - do you think this is worth adding a way to do that In `Expr`?

Reply:
Can we just use IfStmt::getNondiscardedCase here?

More generally, we could add a method for evaluating ConstantExprs, I guess, but IfStmt itself currently just uses EvaluateKnownConstInt, which should be equivalent for the relevant cases.

Reply:
I missed this was a thing. Yes, we should use that!

Reply:
I'm still a bit new here and I'm confused... `CoverageMappingGen.cpp` doesn't have any mention of ASTContext which is an argument for functions mentioned by @efriedma-quic is there an easy way how to get it. Or do I need to go another direction?

Reply:
`CVM.getCodeGenModule().getContext()` from within `CounterCoverageMappingBuilder` might work

Reply:
Ok it seems to work :)

Reply:
Thanks!